pkg/model/trace/matches.go

```diff
@@ -14,21 +14,28 @@ import (
 
 const RootSpanNotYetReceivedText = "<root span not yet received>"
 
+const (
+	attributesNotFound    = 1
+	attributesFailedMatch = 2
+	attributesPassedMatch = 3
+)
+
 func MatchesProto(id []byte, trace *tempopb.Trace, req *tempopb.SearchRequest) (*tempopb.TraceSearchMetadata, error) {
 	traceStart := uint64(math.MaxUint64)
 	traceEnd := uint64(0)
 
-	tagFound := false
+	traceMatch := attributesNotFound
 	if len(req.Tags) == 0 {
-		tagFound = true
+		traceMatch = attributesPassedMatch
 	}
 
 	var rootSpan *v1.Span
 	var rootBatch *v1.ResourceSpans
 	// todo: is it possible to shortcircuit this loop?
 	for _, b := range trace.Batches {
-		if !tagFound && searchAttributes(req.Tags, b.Resource.Attributes) {
-			tagFound = true
+		batchMatch := attributesNotFound
+		if traceMatch != attributesPassedMatch {
+			batchMatch = searchAttributes(req.Tags, b.Resource.Attributes)
```

Comment:
this logic is slightly hard to follow because matching is split between resource attributes and trace attributes. Would it be easier to have a `map[string]bool` initialised with the query parameters, and we can check the state of the map after going through each batch?

Reply:
yeah, agree that this is more complex than i was hoping. having to compare across resource and span as well as having to still iterate spans even if the batch fails a match really adds some complexity.

unsure on your suggestion though. can you provide more details?